lib/internal/bootstrap/node.js

```diff
@@ -108,7 +108,15 @@ process.domain = null;
   ObjectDefineProperty(process, 'exitCode', {
     __proto__: null,
     get() {
-      return exitCode;
+      if (exitCode !== undefined) {
+        return exitCode;
+      }
+
+      if (fields[kHasExitCode] !== 0) {
+        return fields[kExitCode];
+      }
+
+      return undefined;
```

Comment:
I wonder if it makes sense to keep the `exitCode` local variable around at all 🤔
```suggestion
      return fields[kHasExitCode] ? fields[kExitCode] : undefined;
```

Reply:
Well if the tests still pass without it, let's remove it. If a test fail, it would tell us what this variable is for

Reply:
@aduh95 removed, and nothing seemed to blow up 😄 

in the CI the macOS tests failed but that looks very unrelated 🙂

Reply:
Nice, the updated version is much more satisfying in terms of simplicity. And if it was useful for some reason, that's on whoever introduced it for not putting a comment explaining it ;)